python/ray/data/_internal/execution/legacy_compat.py

```diff
@@ -109,6 +116,7 @@ def _get_execution_dag(
     executor: Executor,
     plan: ExecutionPlan,
     allow_clear_input_blocks: bool,
+    preserve_order: bool = False,
```

Comment:
Shall we make it required as other params and as upper call funcs?

Reply:
yeah we can make it required. sorry what do you mean by "as upper call funcs"?

Reply:
Oh, the callers of this function, like `execute_to_legacy_block_list`.

Reply:
Got it, updated.